src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2407,7 +2407,7 @@ internal EffectiveParameters(ImmutableArray<TypeSymbol> types, ImmutableArray<Re
             {
                 int parm = argToParamMap.IsDefault ? arg : argToParamMap[arg];
                 // If this is the __arglist parameter, just skip it.
-                if (parm == parameters.Length)
+                if (parm >= parameters.Length)
```

Comment:
It's not clear to me why, if the parameter length and the argument length are mismatched, the only argument beyond the parameter list will be the _arglist parameter.